internal/fnapi/naming.go

```diff
@@ -83,14 +83,22 @@ func AllocateName(ctx context.Context, opts AllocateOpts) (*NameResource, error)
 
 		var nr IssueResponse
 
+		fetchTenantToken := func(ctx context.Context) (string, error) {
```

Comment:
After moving Naming to tenant tokens, the only remaining calls using user auth tokens are Telemetry and Versions.
Both of these set `OptionalAuth = true` and indeed do not need the token.
So I'd suggest removing the FetchToken from them and updating the type at 
https://github.com/namespacelabs/foundation/blob/0ba4493266b26d72c9a9f2959824e347189139bd/internal/fnapi/fnapi.go#L63-L64
To be `func(ctx context.Context) (*auth.Token, error)`

This would allow us to skip the temp function here and in `rpc.go`